dali/operators/sequence/element_extract.cu

```diff
@@ -18,48 +18,29 @@
 
 namespace dali {
 
-namespace detail {
-
-TensorListShape<> GetOutputShape(const TensorList<GPUBackend> &input,
-                                 const std::vector<int>& element_map) {
-    TensorListShape<> output_shape(input.ntensor(), input.shape().sample_dim() - 1);
-    for (unsigned int i = 0; i < input.ntensor(); ++i) {
-        auto shape = input.tensor_shape(i);
-        CheckInputShape(shape, element_map);
-        output_shape.set_tensor_shape(i, shape.last(shape.size() - 1));
-    }
-    return output_shape;
-}
-
-}  // namespace detail
-
 template <>
 void ElementExtract<GPUBackend>::RunImpl(DeviceWorkspace &ws) {
-    auto &input = ws.Input<GPUBackend>(0);
-    auto output_shape = detail::GetOutputShape(input, element_map_);
-    auto element_layout = VideoLayoutInfo::GetFrameLayout(input.GetLayout());
-    int elements_per_sample = element_map_.size();
-    int output_offset = 0;
-    auto data_type = input.type();
-    for (int k = 0; k < elements_per_sample; k++) {
-        int element = element_map_[k];
-        auto &output = ws.Output<GPUBackend>(output_offset + k);
-        output.set_type(input.type());
-        output.SetLayout(element_layout);
-        output.Resize(output_shape);
-
-        for (unsigned int i = 0; i < input.ntensor(); i++) {
-            auto tensor_shape = input.tensor_shape(i);
-            auto element_size = volume(tensor_shape.begin()+1, tensor_shape.end());
-            auto input_offset_bytes = element * element_size * data_type.size();
-
-            data_type.Copy<GPUBackend, GPUBackend>(
-                output.raw_mutable_tensor(i),
-                static_cast<const uint8_t*>(input.raw_tensor(i)) + input_offset_bytes,
-                element_size,
-                ws.stream());
-        }
+  scatter_gather_.Reset();
```

Comment:
It duplicates most of the CPU RunImpl. Can you extract the common part?

Reply:
Done